### PR TITLE
prepare_release script to ignore some packages/files

### PR DESCRIPTION
**Pull request by owais**
# Description

- Will not ignore files from build/dist directories. The script was picking up version.py files from these directories and updating them instead of the real version.py files.
- Will ignore packages that are released independently of the contrib release process.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Dev/Tooling update.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manual verification

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: scripts/eachdist.py [hunk 1]
```diff
@@ -669,13 +681,16 @@ def release_args(args):
     cfg.read(str(find_projectroot() / "eachdist.ini"))
     versions = args.versions
     updated_versions = []
+
+    excluded = cfg["exclude_release"]["packages"].split()
```
Comment (owais) [inline scripts/eachdist.py:685]: `exclude_release` group from eachdist.ini will now be excluded from this process as they are released independently by their dedicated maintainers.

File: scripts/eachdist.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
 DEFAULT_ALLSEP = " "
 DEFAULT_ALLFMT = "{rel}"
 
+NON_SRC_DIRS = ["build", "dist", "__pycache__", "lib", "venv", ".tox"]
```
Comment (owais) [inline scripts/eachdist.py:20]: when trying to find python source files, we should not look in these directories.

**Review by ocelotl — APPROVED** [review]

**Review by lzchen — APPROVED** [review]

File: eachdist.ini [hunk 3]
```diff
@@ -41,6 +41,12 @@ packages=
     opentelemetry-semantic-conventions
     opentelemetry-test-utils
     opentelemetry-instrumentation
+    opentelemetry-distro
+
+[exclude_release]
+packages=
+    opentelemetry-sdk-extension-aws
+    opentelemetry-propagator-aws-xray
```
Comment (NathanielRN) [inline eachdist.ini:49]: Just curious, don't we already guard ourselves from releasing these packages with this code?

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/3ceab3b51071e50ef74a255b973a73f204c08c8f/scripts/build.sh#L33-L41

Comment (owais) [reply eachdist.ini:49]: This PR fixes the preparation step which bumps package versions and runs locally before the release PR is created.

**Pull request merged**